### PR TITLE
Add WebApiTelemetry extension to identify user and detect incoming requests

### PR DESCRIPTION
**Pull request by PhillipPruett**

### CODE REVIEW

File: Recipes.Tests/WebApiTelemetryTests.cs [hunk 1]
```diff
@@ -138,16 +140,36 @@ public void TearDown()
...
                            .Be("123.123.123.123");
         }
 
         [Test]
+        public async Task Telemetry_events_based_on_HTTP_request_are_marked_as_incoming()
+        {
+            HttpResponseMessage response = null;
+
+            using (Log.With<Telemetry>(t => t.WithPropertiesBasedOn(response)).Enter(() => { }))
+            {
```
Comment (jonsequitur) [inline Recipes.Tests/WebApiTelemetryTests.cs]: This says `WithPropertiesBasedOn(response)` which seems to be different from what the test name says.

File: Recipes.Tests/WebApiTelemetryTests.cs [hunk 2]
```diff
@@ -138,16 +140,36 @@ public void TearDown()
...
-                           .BeTrue();
-            telemetryEvents.Single()
                            .CallerIpAddress()
                            .Should()
                            .Be("123.123.123.123");
         }
 
         [Test]
+        public async Task Telemetry_events_based_on_HTTP_request_are_marked_as_incoming()
+        {
```
Comment (jonsequitur) [inline Recipes.Tests/WebApiTelemetryTests.cs]: Should this be understood to be a different behavior from when it's based on a response object, e.g. what about outgoing calls?

File: Recipes.Tests/WebApiTelemetryTests.cs (hunk 1 above)
Comment (PhillipPruett) [inline Recipes.Tests/WebApiTelemetryTests.cs]: Updated the test name to be more appropriate

File: Recipes.Tests/WebApiTelemetryTests.cs (hunk 2 above)
Comment (PhillipPruett) [inline Recipes.Tests/WebApiTelemetryTests.cs]: correct, withPropertiesBasedOn will set IsIncoming based upon whether or not an action descriptor is present. added a test to show this behavior.

File: Recipes.Tests/WebApiTelemetryTests.cs (hunk 1 above)
Comment (jonsequitur) [inline Recipes.Tests/WebApiTelemetryTests.cs]: I think the test name is still overly broad. It implies all requests based on responses are marked as incoming. The next test has a "when" and I think this one needs one as well.

Comment (PhillipPruett) [inline Recipes.Tests/WebApiTelemetryTests.cs]: done

**Pull request merged**